hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/RowIndexSeekerV1.java

```diff
@@ -131,8 +131,7 @@ private int binarySearch(Cell seekCell, boolean seekBefore) {
     int comp = 0;
     while (low <= high) {
       mid = low + ((high - low) >> 1);
-      ByteBuffer row = getRow(mid);
-      comp = compareRows(row, seekCell);
+      comp = this.cellComparator.compareRows(getRow(mid), seekCell);
```

Comment:
I too agree to @anoopsjohn . Seems we deliberately left this compareRows() here. Is there a similar compareRows in every encoder?

Reply:
There is a similar array-based compare rows in CellComparator.

See above for my argument that encoders should not each have to do their own figuring of cell compare.

Notice how much cleaner the code in here is now?